packages/costa/src/utils.ts

```diff
@@ -17,3 +17,17 @@ export function pipe(readable: NodeJS.ReadableStream, writable: NodeJS.WritableS
     writable.write(data);
   });
 }
+
+/**
+ * pipe read stream to write steam with finish event
+ * @param readStream the read stream
+ * @param writeStream the write stream
+ */
+export function pipeWithFinish(readStream: NodeJS.ReadableStream, writeStream: NodeJS.WritableStream): Promise<void> {
```

Comment:
The name looks wired a bit, `finish` is just an event.

Reply:
How about `pipePromisify`?

Reply:
I think `safePipe` is more specific.

Reply:
Or `pipeGracefully`?

Reply:
pipeGracefully +1